### PR TITLE
[Doc] Initial commit to generate doccument using hotdoc @open sesame 01/21 16:10

### PR DESCRIPTION
**Pull request by gichan-jang**
Initial commit to generate doccument using hotdoc.

 https://nnstreamer.github.io/index.html

Related issue: #2860 

Genetated HTML examples:
mainpage
edgeai
example

Signed-off-by: Gichan Jang <gichan2.jang@samsung.com>

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed []Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -508,3 +508,31 @@ if get_option('enable-test')
...
+      project_version: meson.project_version(),
+      sitemap: 'sitemap.txt',
+      index: 'index.md',
+      install: true,
+      syntax_highlighting_activate: true,
+      html_extra_theme: join_paths(meson.current_source_dir(), 'theme/extra'),
+      disable_incremental_build: true,
+      build_always_stale: true,
+  )
+endif
```
Comment (dongju-chae) [inline meson.build:538]: How about moving this and related files into `./Documentation`? e.g., `Documentation/meson.build`

File: Documentation/component-description.md [hunk 2]
```diff
@@ -28,7 +32,7 @@ In this page, we focus on the status of each elements. For requirements and desi
     - application/octet-stream. Stream pipeline developer MUST specify the corresponding type and dimensions via properties (input-dim, input-type)
   - You may add subplugins of converters. However, as of 2020-01-10, we do not have any converter subplugins released although we do support them. Users may add such subplugins in run-time.
     - WIP: flatbuffers, protobuf
-- [tensor\_filter](../gst/nnstreamer/tensor_filter)
+- [tensor\_filter](.https://github.com/nnstreamer/nnstreamer/tree/main/gst/nnstreamer/tensor_filter)
```
Comment (anyj0527) [inline Documentation/component-description.md:35]: This `.` might be a typo

File: meson.build (hunk 1 above)
Comment (gichan-jang) [reply meson.build:538]: I agree with some of your opinions.
However, The source code such as `gst` and `ext` will be parsed and the API documents will be generated later.
So, I think it's better to put it here.

File: Documentation/component-description.md (hunk 2 above)
Comment (gichan-jang) [reply Documentation/component-description.md:35]: Thanks, I will update after solving CI problem.

File: meson.build (hunk 1 above)
Comment (gichan-jang) [reply meson.build:519]: Since API reference is just linked doxygen book, I moved meson command as you suggested. Thanks.

**Comment by gichan-jang** [comment]
- NNStreamer Element description, NNStreamer Example, Doxygen book link(API reference) are added.

File: Documentation/meson.build [hunk 3]
```diff
@@ -4,6 +4,8 @@ if not hotdoc_p.found()
     subdir_done()
 endif
 
+run_command('git', 'clone', 'https://github.com/nnstreamer/nnstreamer-example', 'nnst-exam')
```
Comment (dongju-chae) [inline Documentation/meson.build:7]: Need to check whether it exists or not. BTW, is this used only internally? `pdebuild` or `gbs` cannot run this command.

Comment (gichan-jang) [reply Documentation/meson.build]: I changed the hotdoc document generation from meson build to bash script.
Users who want to generate documents can use the scripts.

**Review by dongju-chae — APPROVED** [review]
LGTM

File: Documentation/component-description.md [hunk 4]
```diff
@@ -55,20 +59,20 @@ In this page, we focus on the status of each elements. For requirements and desi
   - SNPE (stable)
   - Vivante (stable)
   - WIP: SNAP (Exynos-NPU & Qualcomm-SNPE), ...
-  - [Guide on writing a filter subplugin](writing-subplugin-tensor-filter.md)
+  - [Guide on writing a filter subplugin](Documentation/writing-subplugin-tensor-filter.md)
```
Comment (jaeyun-jung) [inline Documentation/component-description.md:62]: this change will break the link

**Review by jaeyun-jung — APPROVED** [review]

**Review by myungjoo — APPROVED** [review]

**Pull request merged**